### PR TITLE
Show proper error from init process panic

### PR DESCRIPTION
**Pull request by tonistiigi**
@crosbymichael @mlaventure 

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>

### CODE REVIEW

File: libcontainer/factory_linux.go [hunk 1]
```diff
@@ -248,6 +249,13 @@ func (l *LinuxFactory) StartInitialization() (err error) {
 		// ensure that this pipe is always closed
 		pipe.Close()
 	}()
+
+	defer func() {
+		if e := recover(); e != nil {
+			err = fmt.Errorf("panic from initialization: %v, %v", e, string(debug.Stack()))
```
Comment (cyphar) [inline libcontainer/factory_linux.go]: Why not just use the passed argument to `panic`, rather than `debug.Stack`?

**Comment by mlaventure** [comment]
:+1:, this could prove very useful

**Comment by crosbymichael** [comment]
@cyphar is this a better way to use the error now?

**Comment by hqhq** [comment]
Very helpful, LGTM

**Comment by crosbymichael** [comment]
LGTM

**Comment by cyphar** [comment]
LGTM. Hopefully the stack traces won't be too noisy.

**Comment by mrunalp** [comment]
LGTM

**Pull request merged**